what4/src/What4/Utils/BVDomain/Arith.hs

```diff
@@ -276,6 +279,55 @@ ult a b
     (a_min, a_max) = ubounds a
     (b_min, b_max) = ubounds b
 
+-- | Check if @(bvult (bvadd a c) (bvadd b c))@ is equivalent to @(bvult a b)@.
+-- 
+-- This is true if and only if for all integer values @i_a@, @i_b@, @i_c@ in
+-- @a@, @b@, @c@, either both @(bvadd i_a i_c)@ and @(bvadd i_b i_c)@ wrap
+-- around, or both do not. We prove this by contradiction. Assume that @(bvult
```

Comment:
You define `i_a`, `i_b`, and `i_c` to be integer values, but you also pass them to `bvadd`, which expects bitvectors, not integers. As a result, I find it difficult to read this comment, as I'm not sure which things are bitvectors (i.e., wrapped modulo `2^w`) and which things are integers (i.e., unbounded).